tests/flows/staking/poa_validator_manager.go

```diff
@@ -0,0 +1,206 @@
+		},
+	)
+
+	// Generate random address to be the owner address
+	ownerKey, err := crypto.GenerateKey()
+	Expect(err).Should(BeNil())
+	ownerAddress := crypto.PubkeyToAddress(ownerKey.PublicKey)
+
+	// Transfer native assets to the non owner account
+	ctx := context.Background()
+	fundAmount := big.NewInt(1e18) // 1avax
+	utils.SendNativeTransfer(
+		ctx,
+		subnetAInfo,
+		fundedKey,
+		ownerAddress,
+		fundAmount,
+	)
+
+	validatorManagerAddress, validatorManager := utils.DeployAndInitializePoAValidatorManager(
+		ctx,
+		fundedKey,
+		subnetAInfo,
+		pChainInfo,
+		ownerAddress,
+	)
+
+	var validationID ids.ID // To be used in the delisting step
+	nodeID := ids.GenerateTestID()
+	blsPublicKey := [bls.PublicKeyLen]byte{}
+	weight := uint64(1)
+
+	{
+		// Try to call with invalid owner
+		opts, err := bind.NewKeyedTransactorWithChainID(fundedKey, subnetAInfo.EVMChainID)
+		Expect(err).Should(BeNil())
+
+		_, err = validatorManager.InitializeValidatorRegistration(
+			opts,
+			weight,
+			nodeID,
+			uint64(time.Now().Add(24*time.Hour).Unix()),
+			blsPublicKey[:],
+		)
+		Expect(err).ShouldNot(BeNil())
+
+		// Initiate validator registration
+		var receipt *types.Receipt
+		receipt, validationID = utils.InitializePoAValidatorRegistration(
+			ownerKey,
+			subnetAInfo,
+			weight,
+			nodeID,
+			blsPublicKey,
+			validatorManager,
+		)
+
+		// Gather subnet-evm Warp signatures for the RegisterSubnetValidatorMessage & relay to the P-Chain
+		// (Sending to the P-Chain will be skipped for now)
+		signedWarpMessage := network.ConstructSignedWarpMessage(context.Background(), receipt, subnetAInfo, pChainInfo)
+
+		// Validate the Warp message, (this will be done on the P-Chain in the future)
+		utils.ValidateRegisterSubnetValidatorMessage(
+			signedWarpMessage,
+			nodeID,
+			weight,
+			subnetAInfo.SubnetID,
+			blsPublicKey,
+		)
+
+		// Construct a SubnetValidatorRegistrationMessage Warp message from the P-Chain
+		registrationSignedMessage := utils.ConstructSubnetValidatorRegistrationMessage(
+			validationID,
+			true,
+			subnetAInfo,
+			pChainInfo,
+			network,
+			signatureAggregator,
+		)
+
+		// Deliver the Warp message to the subnet
+		receipt = utils.CompletePoAValidatorRegistration(
+			fundedKey,
+			subnetAInfo,
+			validatorManagerAddress,
+			registrationSignedMessage,
+		)
+		// Check that the validator is registered in the staking contract
+		registrationEvent, err := utils.GetEventFromLogs(
+			receipt.Logs,
+			validatorManager.ParseValidationPeriodRegistered,
+		)
+		Expect(err).Should(BeNil())
+		Expect(registrationEvent.ValidationID[:]).Should(Equal(validationID[:]))
+	}
+
+	//
+	// Delist the validator
+	//
+	{
+		// Try with invalid validator owner
+		opts, err := bind.NewKeyedTransactorWithChainID(fundedKey, subnetAInfo.EVMChainID)
+		Expect(err).Should(BeNil())
+		_, err = validatorManager.InitializeEndValidation(
+			opts,
+			validationID,
+		)
+
+		Expect(err).ShouldNot(BeNil())
```

Comment:
```suggestion
		Expect(err).ShouldNot(BeNil())
		
```
associate the error check with the function call